src/flow.c

```diff
@@ -453,6 +453,14 @@ void FlowHandlePacketUpdate(Flow *f, Packet *p)
             f->flags &= ~FLOW_PROTO_DETECT_TS_DONE;
             p->flags |= PKT_PROTO_DETECT_TS_DONE;
         }
+        if (MacSetFlowStorageEnabled()) {
```

Comment:
`if (p->ethh && MacSetFlowStorageEnabled())` to avoid a small check and the following `FlowGetStorageById` check for non-ethernet.

Reply:
Would also like to see this logic move into a small `inline` util func to keep this code as clean as possible

Reply:
> `if (p->ethh && MacSetFlowStorageEnabled())` to avoid a small check and the following `FlowGetStorageById` check for non-ethernet.

Sure, I can pull up the `p->ethh` NULL check to the outer `if` confdition but will need to keep the check for the result of `FlowGetStorageById` being NULL, right?
My idea:
```C
static inline void FlowUpdateEthernet(Flow *f, EthernetHdr *ethh, ThreadVars *tv,
                                      DecodeThreadVars *dtv, bool toserver)
{
    if (ethh && MacSetFlowStorageEnabled()) {
        MacSet *ms = FlowGetStorageById(f, MacSetGetFlowStorageID());
        if (ms != NULL) {
            if (toserver) {
                MacSetAddWithCtr(ms, ethh->eth_src, ethh->eth_dst, tv,
                                 dtv->counter_max_mac_addrs_src,
                                 dtv->counter_max_mac_addrs_dst);
            } else {
                MacSetAddWithCtr(ms, ethh->eth_dst, ethh->eth_src, tv,
                                 dtv->counter_max_mac_addrs_dst,
                                 dtv->counter_max_mac_addrs_src);
            }
        }
    }
}
```

Reply:
Yes that looks sane. The `FlowGetStorageById` result needs to be checked indeed.

Minor style comment: our convention is to have `tv` and `dtv` as the first to args of the func.

Reply:
OK, got it.